src/piece_picker.cpp

```diff
@@ -2921,6 +2921,26 @@ namespace libtorrent
 		return true;
 	}
 
+	bool piece_picker::is_piece_ready(int index) const
+	{
+		if (!is_piece_finished(index))
+			return false;
+
+		piece_pos const& p = m_piece_map[index];
+		int state = p.download_queue();
+
+		if (state < piece_pos::num_download_categories)
+		{
+			std::vector<downloading_piece>::const_iterator i =
+				find_dl_piece(state, index);
+			TORRENT_ASSERT(i != m_downloads[state].end());
+			if (int(i->finished) < blocks_in_piece(index))
+				return false;
```

Comment:
I think you need to checkt ``i->passed_hash_check`` here too. But I could be wrong. If I am, maybe there should be a comment here explaining why